src/main/java/bio/terra/stairway/Flight.java

```diff
@@ -169,13 +170,23 @@ private StepResult stepWithRetry() throws InterruptedException {
 
         // Retry loop
         do {
-            // Do or undo based on direction we are headed
-            if (context().isDoing()) {
-                result = currentStep.step.doStep(context());
-            } else {
-                result = currentStep.step.undoStep(context());
+            try {
+                // Do or undo based on direction we are headed
+                if (context().isDoing()) {
+                    result = currentStep.step.doStep(context());
+                } else {
+                    result = currentStep.step.undoStep(context());
+                }
 
+            } catch (Exception ex) {
+                // The purpose of this catch is to relieve steps of implementing their own repetitive try-catch
+                // simply to turn exceptions into StepResults.
+                StepStatus stepStatus = (ex instanceof RetryException)
```

Comment:
Where does a RetryException actually get thrown? I don't think the individual Step should throw it. I only saw it used in a test fixture.

Reply:
My idea is that steps could throw exceptions derived from RetryException and Stairway would retry. The other way available to steps is to return StatusResult with the RETRY status.

If when we implement retrying and we don't use the exception, then I can remove it.

Reply:
Since we aren't forcing the Step code to handle exceptions, I thought it would be the flight code that would decide to do the retry if it caught an exception.

Reply:
Without something telling it the difference, Stairway won't know which exceptions to retry and which to undo. With this PR, there are two ways we could code things:

1. For steps that will retry, they will have to implement exception handling for the retryable exceptions and return the StatusResult with RETRY status.

2. Or in the DAO and PDAO, we catch failures we want to retry and throw exceptions derived from RetryException. That let's stairway know that it should retry instead of undo-ing.

I think we will find that #2 is easier to deal with, but since we haven't done much coding for retry, I'm not sure which we will prefer.